src/publishProject/publishProjectWebViewController.ts

```diff
@@ -806,9 +851,33 @@ export class PublishProjectWebViewController extends FormWebviewController<
             this.state.formState,
         );
 
+        // Check SQLCMD variables validation
+        const sqlCmdVariablesValid = this.allSqlCmdVariablesFilled();
+
         // hasFormErrors state tracks to disable buttons if ANY errors exist
-        this.state.hasFormErrors = hasValidationErrors || hasMissingRequiredValues;
+        this.state.hasFormErrors =
+            hasValidationErrors || hasMissingRequiredValues || !sqlCmdVariablesValid;
 
         return erroredInputs;
     }
+
+    /**
+     * Validates that all SQLCMD variables have non-empty values
+     * This follows the same validation logic as the SQL projects extension
+     */
+    private allSqlCmdVariablesFilled(): boolean {
+        const sqlCmdVariables = this.state.formState.sqlCmdVariables;
+        if (!sqlCmdVariables) {
+            return true; // No variables to validate
+        }
+
+        for (const varName in sqlCmdVariables) {
+            const value = sqlCmdVariables[varName];
+            if (value === "" || value === undefined) {
```

Comment:
what about empty string? and if there's already a validation somewhere, why not using a shared code?

Reply:
the comment is misleading, it was added to tell that the logic is similar to ADS sql projects validation, which can be removed.  also it is checking for the empty sting already